### PR TITLE
go.mod: bump versions

### PR DESCRIPTION
**Pull request by howardjohn**
# Changes

This PR bumps various go.mod dependencies. The intent here is I indirectly import this repo and the mix of dependency versions is causing (minor) problems.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- NA: Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps

- NA: Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

/kind cleanup

### CODE REVIEW

**Comment by howardjohn** [comment]
Note the test changes are due to https://github.com/google/go-containerregistry/commit/a0cca8a048d05579c7107708862fe8e6f132efbf

**Review by vdemeester — CHANGES_REQUESTED** [review]
@howardjohn thanks for the PR, one comment 👼🏼

File: go.mod [hunk 1]
```diff
@@ -6,35 +6,35 @@ require (
...
-	k8s.io/code-generator v0.25.9
+	k8s.io/api v0.27.2
+	k8s.io/apimachinery v0.27.2
+	k8s.io/client-go v0.27.2
+	k8s.io/code-generator v0.27.2
 	k8s.io/klog v1.0.0
-	k8s.io/kube-openapi v0.0.0-20230308215209-15aac26d736a
-	knative.dev/pkg v0.0.0-20230221145627-8efb3485adcf
+	k8s.io/kube-openapi v0.0.0-20230515203736-54b630e78af5
+	knative.dev/pkg v0.0.0-20230612155445-74c4be5e935e
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply go.mod:37]: Is there a source you use for min k8s version? https://github.com/kubernetes/client-go#compatibility-client-go---kubernetes-clusters seems not updated. Looks like the current min k8s version is 1.24 (https://tekton.dev/docs/pipelines/resolution-getting-started/#prerequisites), IME client-go 1.27 works perfectly fine with k8s 1.24 but not sure the process typically used here

[the remark this answers is not in the document]
Comment (vdemeester) [reply go.mod:37]: @howardjohn we mainly look at this https://github.com/knative/pkg/blob/main/version/version.go#L36 for setting our own kubernetes min version.

(Looks like main pulls 0.26.x https://github.com/knative/pkg/blob/main/go.mod#L46)

[the remark this answers is not in the document]
Comment (howardjohn) [reply go.mod]: Ok coming back (months later). Knative released 1.11, so this updates to that version. Knative 1.11 has minimum version 1.25: https://github.com/knative/pkg/blob/release-1.11/version/version.go#L36.

1.10 had 1.24, so I think this bumps tekton minimum to 1.25 as well?

[the remark this answers is not in the document]
Comment (vdemeester) [reply go.mod]: @howardjohn See https://github.com/tektoncd/pipeline/pull/6975 (and https://github.com/tektoncd/plumbing/pull/1520) 😅

**Comment by howardjohn** [comment]
https://github.com/tektoncd/pipeline/pull/6975 handles this

**Pull request closed without merging**